x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/aggregate/SpatialAggregateFunction.java

```diff
@@ -22,26 +25,34 @@
  * select the best one.
  */
 public abstract class SpatialAggregateFunction extends AggregateFunction {
-    protected final boolean useDocValues;
+    protected final FieldExtractPreference preference;
 
-    protected SpatialAggregateFunction(Source source, Expression field, Expression filter, boolean useDocValues) {
+    protected SpatialAggregateFunction(Source source, Expression field, Expression filter, FieldExtractPreference values) {
         super(source, field, filter, emptyList());
-        this.useDocValues = useDocValues;
```

Comment:
Given this change here, I'm wondering if there are no serialization issues regarding BWC.

Reply:
Could you please elaborate?

Reply:
Some things are serialized in communication node to node. I am wondering if the change from boolean to FieldExtractPreference has any connection with serialization and, implicitly, BWC (backwards compatibility). Have you looked into that?

Reply:
Given the comment in the file:
```
        // The useDocValues field is only used on data nodes local planning, and therefor never serialized
```
Wouldn't this mean the isn't any issue regarding serialization?

Reply:
Correct. This field is only every planned or modified on the data nodes, and is therefor never serialized.